src/main/scala/com/campudus/tableaux/database/model/structure/ColumnModel.scala

```diff
@@ -40,7 +40,17 @@ class ColumnModel(val connection: DatabaseConnection) extends DatabaseQuery {
 
       case CreateLinkColumn(name, orderingOpt, toTableId, toName, singleDirection, identifier, di) => for {
         toTable <- tableStruc.retrieve(toTableId)
-        toCol <- retrieveAll(toTable).map(_.head.asInstanceOf[ValueColumn[_]])
+        toTableColumns <- retrieveAll(toTable).flatMap({
```

Comment:
I think you can do something like this to remove the `flatMap`:

```
       toTableColumns <- retrieveAll(toTable) if (!toTableColumns.isEmpty)
```

If you want to keep the `NotFoundInDatabaseException`, you could then do a `recoverWith` or put it into another `<-` instead of the `flatMap`.